### DISCUSSION TITLE
Send externally signed transaction.

### DISCUSSION THREAD

**Discussion by ghost** [wevm/viem#785] [Idea / Feature Request] [answered]
Good afternoon,

I would like to know if it is possible for you to allow broadcasting transactions that have been signed externally.

In my use case, the private keys are stored on secure hardware (https://fr.wikipedia.org/wiki/Hardware_Security_Module) and that hardware manages signing transactions.

For now, I am using ethers v5 and ethereumjs/tx to broadcast transactions externally signed using the following  https://docs.ethers.org/v5/api/providers/provider/#Provider-sendTransaction

Is it planned to add that kind of feature?

**Comment by milaabl** [comment]
@Fbartoli , wouldn't something like that work for you?

```
const hash = await walletClient.sendTransaction({ 
  account,
  to: '0x70997970c51812dc3a010c7d01b50e0d17dc79c8',
  value: 1000000000000000000n
})
```

**Reply by ghost** [reply]
Hey, no I need something like 
`await provider.send("eth_sendRawTransaction", [
     signedTx,
  ]);`

**Comment by quagliero** [comment]
Came here to ask the exact same question, hah.

In my situation I am currently `ethers` to sign a tx from a dynamically created wallet and then passing it to multiple RPCs to send via:
```ts
 const txHash = await provider.send("eth_sendRawTransaction", [
     signedTx,
  ]);
 ```
 
 I've been refactoring to use viem instead of ethers but just hit a stumbling block where I've found that `walletClient.signTransaction` isn't supported yet and I don't seem to be able to broadcast a tx directly on the publicClient?
 
 Am I missing something?

**Comment by jxom** [comment, marked as answer]
Yeah, we will need to implement sendRawTransaction. Shouldn’t take too long. Can look into it tomorrow.

**Reply by ghost** [reply]
Thank you a lot, we are updating our packages and this is an important point for us, thanks a lot.

**Comment by ghost** [comment]
Hello, sorry to ask.

Do you have a date for the update? 
May I help with a merge request?

**Reply by jxom** [reply]
Have a few things to do first before this. Will let you know when it’s ready!

**Reply by varunsrin** [reply]
plus one from us on this request, would help the Warpcast backend eliminate another dependency

**Reply by jxom** [reply]
Coming: https://github.com/wagmi-dev/viem/pull/1058
